src/app.controller.ts

```diff
@@ -46,4 +65,37 @@ export class AppController {
     );
     //TODO Once this completes publish the credential share complete with the interaction id to the client
   }
+
+  @Get(RestEndpoint.GEO_LOCATION)
+  public async getGeo(
+    @Req() req: Request
+  ): Promise<GeoInformation | undefined> {
+    const forwardedFor = req.headers['x-forwarded-for'];
+
+    if (!forwardedFor) {
+      this.logger.error('X-Forwarded-For header not set');
+      return undefined;
+    }
+
+    const ip = String(forwardedFor).split(',')?.[0];
```

Comment:
Are you sure it's the first and not the last item in the array?

Reply:
That's what I understood from https://stackoverflow.com/a/72586833

Reply:
I think you are right, but we need to check.